classroom/snippets/src/main/java/ListStudentSubmissions.java

```diff
@@ -0,0 +1,90 @@
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.services.classroom.Classroom;
+import com.google.api.services.classroom.ClassroomScopes;
+import com.google.api.services.classroom.model.ListStudentSubmissionsResponse;
+import com.google.api.services.classroom.model.StudentSubmission;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/* Class to demonstrate the use of Classroom List StudentSubmissions API. */
+public class ListStudentSubmissions {
+  /**
+   * Lists student submissions based on courseId, courseWorkId, and userId.
+   *
+   * @param courseId - identifier of the course.
+   * @param courseWorkId - identifier of the course work.
+   * @param userId - identifier of the student whose work to return.
+   * @return - list of student submissions.
+   * @throws IOException - if credentials file not found.
+   */
+  public static List<StudentSubmission> listStudentSubmissions(String courseId, String courseWorkId,
+      String userId) throws IOException {
+    /* Load pre-authorized user credentials from the environment.
+     TODO(developer) - See https://developers.google.com/identity for
+      guides on implementing OAuth2 for your application. */
+    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault()
+        .createScoped(Collections.singleton(ClassroomScopes.CLASSROOM_COURSEWORK_STUDENTS));
+    HttpRequestInitializer requestInitializer = new HttpCredentialsAdapter(
+        credentials);
+
+    // Create the classroom API client.
+    Classroom service = new Classroom.Builder(new NetHttpTransport(),
+        GsonFactory.getDefaultInstance(),
+        requestInitializer)
+        .setApplicationName("Classroom samples")
+        .build();
+
+    // [START classroom_list_student_submissions_code_snippet]
+
+    List<StudentSubmission> studentSubmissions = new ArrayList<>();
+    try {
+      // Set the userId as a query parameter on the request.
+      ListStudentSubmissionsResponse response = service.courses().courseWork().studentSubmissions()
+          .list(courseId, courseWorkId)
+          .set("userId", userId)
+          .execute();
+      studentSubmissions.addAll(response.getStudentSubmissions());
+    } catch (GoogleJsonResponseException e) {
+      //TODO (developer) - handle error appropriately
+      GoogleJsonError error = e.getDetails();
+      if (error.getCode() == 404) {
+        System.out.printf("The courseId (%s), courseWorkId (%s), or userId (%s) does "
+            + "not exist.\n", courseId, courseWorkId, userId);
+      } else {
+        throw e;
+      }
+    } catch (Exception e) {
+      throw e;
+    }
+    return studentSubmissions;
```

Comment:
This is a general observation, and definitely outside the scope of achieving parity between Java and the existing examples, but maybe something to think about in the future.

I see in the existing version of this that it also just returns the list of submissions. I *think* that in the Python version this is a list of dictionaries representing returned JSON. Is this also the case for Java?

I think we might want to instead show an unpacked, printed result in these examples. As-is, we're asking the reader to take extra steps to dive into the structure of the StudentSubmission object; perhaps we should instead demonstrate its contents with a small loop that prints the submission details.